src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,46 +496,67 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
      * </pre>
      *
      * @param currentCase AST of the case that falls through to the next case.
-     * @param nextCase AST of the next case.
      * @return True if a relief comment was found
      */
-    private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase) {
-        boolean allThroughComment = false;
-        final int endLineNo = nextCase.getLineNo();
-
-        if (matchesComment(reliefPattern, endLineNo)) {
-            allThroughComment = true;
+    private boolean hasFallThroughComment(DetailAST currentCase) {
+        final DetailAST nextCase = currentCase.getNextSibling();
+        final DetailAST ast;
+        final DetailAST lastAST;
+        if (nextCase.getType() == TokenTypes.CASE_GROUP) {
+            ast = nextCase;
+            lastAST = nextCase.getLastChild();
```

Comment:
from current case we get next case and then we get last node of next case.

sounds strange..... we should work on current case or previous case.

Please fix wording of variable names to make it very clear what is  inside of variables. like "startAst", "stopAst"

Reply:
done